test/hotspot/jtreg/compiler/lib/template_framework/CodeFrame.java

```diff
@@ -29,22 +29,98 @@
 import java.util.List;
 
 /**
- * The {@link CodeFrame} represents a frame (i.e. scope) of code, appending {@link Code} to the {@code 'codeList'}
+ * The {@link CodeFrame} represents a frame (i.e. scope) of generated code by appending {@link Code} to the {@code 'codeList'}
  * as {@link Token}s are rendered, and adding names to the {@link NameSet}s with {@link Template#addStructuralName}/
- * {@link Template#addDataName}. {@link Hook}s can be added to a frame, which allows code to be inserted at that
- * location later. When a {@link Hook} is {@link Hook#anchor}ed, it separates the Template into an outer and inner
- * {@link CodeFrame}, ensuring that names that are added inside the inner frame are only available inside that frame.
+ * {@link Template#addDataName}. {@link Hook}s can be added to a code frame, which allows code to be inserted at that
+ * location later.
  *
  * <p>
- * On the other hand, each {@link TemplateFrame} represents the frame (or scope) of exactly one use of a
- * Template.
+ * The {@link CodeFrame} thus implements the {@link Name} non-transparency aspect of {@link ScopeToken}.
  *
  * <p>
- * For simple Template nesting, the {@link CodeFrame}s and {@link TemplateFrame}s overlap exactly.
- * However, when using {@link Hook#insert}, we simply nest {@link TemplateFrame}s, going further "in",
- * but we jump to an outer {@link CodeFrame}, ensuring that we insert {@link Code} at the outer frame,
- * and operating on the names of the outer frame. Once the {@link Hook#insert}ion is complete, we jump
- * back to the caller {@link TemplateFrame} and {@link CodeFrame}.
+ * The {@link CodeFrame}s are nested relative to the order of the final rendered code. This can
+ * diverge from the nesting order of the {@link Template} when using {@link Hook#insert}, where
+ * the execution jumps from the current (caller) {@link CodeFrame} scope to the scope of the
+ * {@link Hook#anchor}. This ensures that the {@link Name}s of the anchor scope are accessed,
+ * and not of the ones from the caller scope. Once the {@link Hook#insert}ion is complete, we
+ * jump back to the caller {@link CodeFrame}.
+ *
+ * <p>
+ * Note, that {@link CodeFrame}s and {@link TemplateFrame}s often go together. But they do diverge when
+ * we call {@link Hook#insert}. On the {@link CodeFrame} side, the inserted scope is nested in the anchoring
+ * scope, so that the inserted scope has access to the Names of the anchoring scope, and not the caller
+ * scope. But the {@link TemplateFrame} of the inserted scope is nested in the caller scope, so
+ * that the inserted scope has access to hashtag replacements of the caller scope, and not the
+ * anchoring scope.
+ */
+
+/*
+ * Below, we look at a few examples, and show the use of CodeFrames (c) and TemplateFrames (t).
+ *
+ * Example1: anchoring and insertion in the same Template
```

Comment:
I think this example is already powerful enough to get a feeling of what is going on - thanks for adding it! From my side, no more examples are required for now but feel free to add more examples as you see fit. 

If we stick with one example:

```suggestion
 * Below, we look at an example, and show the use of CodeFrames (c) and TemplateFrames (t).
```